### PR TITLE
Adding blob support for TransactionRequest 

### LINKED ISSUE #5118: add blob support in TransactionRequest
### Describe the feature

Currently the imp of TransactionRequest doesn't have blob support
https://github.com/paradigmxyz/reth/blob/33df4e57664425a3df06fd8f06a21c1526257f40/crates/rpc/rpc-types/src/eth/transaction/request.rs#L46

Todo : 

Adding support for blob

I'am happy to take this one :) 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
This PR need https://github.com/paradigmxyz/reth/pull/5121 this PR to be able to build since i changed u128 to U128

this PR should be able to close https://github.com/paradigmxyz/reth/issues/5118

### CODE REVIEW

File: crates/rpc/rpc-types/src/eth/transaction/request.rs [hunk 1]
```diff
@@ -39,6 +38,14 @@ pub struct TransactionRequest {
...
     #[serde(rename = "type")]
     pub transaction_type: Option<U8>,
+
+    /// Max Fee per Blob gas for EIP-4844 transactions
+    pub max_fee_per_blob_gas: Option<U128>,
+
+    ///
+    pub blob_versioned_hashes: Option<Vec<B256>>,
+    ///
+    pub sidecar: Option<BlobTransactionSidecar>,
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/transaction/request.rs:48]: needs docs

Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/transaction/request.rs:48]: this should be marked with serde(flattened)

File: crates/rpc/rpc-types/src/eth/transaction/request.rs [hunk 2]
```diff
@@ -94,7 +104,7 @@ impl TransactionRequest {
                 }))
             }
             // EIP1559
-            (None, Some(_), access_list) | (None, None, access_list @ None) => {
+            (None, Some(_), access_list) => {
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/transaction/request.rs:107]: this is not correct, this should enforce that all eip4844 fields are none

File: crates/rpc/rpc-types/src/eth/transaction/request.rs [hunk 3]
```diff
@@ -111,7 +121,91 @@ impl TransactionRequest {
                     access_list: access_list.unwrap_or_default(),
                 }))
             }
+            // EIP4844
+            (None, None, access_list) => {
```
Comment (mattsse) [inline crates/rpc/rpc-types/src/eth/transaction/request.rs:125]: this needs to enforce that all 4844 fields are present

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
thanks, I'll merge these into #5193 because there are some conflicts

**Comment by DoTheBestToGetTheBest** [comment]
> thanks, I'll merge these into #5193 because there are some conflicts

oh but i didn't finish this

**Comment by mattsse** [comment]
right, sorry, this was closed premature.

there'll be some conflicts after #5193 (function moved to compat)

**Comment by DoTheBestToGetTheBest** [comment]
> right, sorry, this was closed premature.
> 
> there'll be some conflicts after #5193 (function moved to compat)

oh okey, don't worry then, i understand , 

thank you for this issue !

i'm still working on the new issue where i'am assigned, just it take some time because i'm facing some problems with this issue, I hope the deadline is not very close

**Comment by mattsse** [comment]
sgtm!

no deadline, but feel free to open a draft anytime so I can provider some pointers more easily

**Comment by DoTheBestToGetTheBest** [comment]
> sgtm!
> 
> no deadline, but feel free to open a draft anytime so I can provider some pointers more easily

well tbh i don't want to open draft for this because i know what to do for it but i'm just blocked in something which i want to resolve it alone, don't wanna bother you with this

Thank you so much for all help and pointers you're giving for all this community !

**Pull request closed without merging**